doc/sphinx/language/coq-library.rst

```diff
@@ -1,1112 +1,988 @@
-
-\subsubsection[Programming]{Programming\label{Programming}
-\index{Programming}
-\label{libnats}
-\ttindex{unit}
-\ttindex{tt}
-\ttindex{bool}
-\ttindex{true}
-\ttindex{false}
-\ttindex{nat}
-\ttindex{O}
-\ttindex{S}
-\ttindex{option}
-\ttindex{Some}
-\ttindex{None}
-\ttindex{identity}
-\ttindex{refl\_identity}}
-
-\begin{coq_example*}
-Inductive unit : Set := tt.
-Inductive bool : Set := true | false.
-Inductive nat : Set := O | S (n:nat).
-Inductive option (A:Set) : Set := Some (_:A) | None.
-Inductive identity (A:Type) (a:A) : A -> Type :=
+.. index::
+  single: absurd (term)
+  single: eq_sym (term)
+  single: eq_trans (term)
+  single: f_equal (term)
+  single: sym_not_eq (term)
+  single: eq_ind_r (term)
+  single: eq_rec_r (term)
+  single: eq_rect (term)
+  single: eq_rect_r (term)
+
+.. coqtop:: in
+
+  Theorem absurd : forall A C:Prop, A -> ~ A -> C.
+  Section equality.
+  Variables A B : Type.
+  Variable f : A -> B.
+  Variables x y z : A.
+  Theorem eq_sym : x = y -> y = x.
+  Theorem eq_trans : x = y -> y = z -> x = z.
+  Theorem f_equal : x = y -> f x = f y.
+  Theorem not_eq_sym : x <> y -> y <> x.
+  End equality.
+  Definition eq_ind_r :
+   forall (A:Type) (x:A) (P:A->Prop), P x -> forall y:A, y = x -> P y.
+  Definition eq_rec_r :
+   forall (A:Type) (x:A) (P:A->Set), P x -> forall y:A, y = x -> P y.
+  Definition eq_rect_r :
+   forall (A:Type) (x:A) (P:A->Type), P x -> forall y:A, y = x -> P y.
+  Hint Immediate eq_sym not_eq_sym : core.
+
+.. index::
+  single: f_equal2 ... f_equal5 (term)
+
+The theorem ``f_equal`` is extended to functions with two to five
+arguments. The theorem are names ``f_equal2``, ``f_equal3``, 
+``f_equal4`` and ``f_equal5``.
+For instance ``f_equal3`` is defined the following way.
+
+.. coqtop:: in
+  
+  Theorem f_equal3 :
+   forall (A1 A2 A3 B:Type) (f:A1 -> A2 -> A3 -> B)
+     (x1 y1:A1) (x2 y2:A2) (x3 y3:A3),
+     x1 = y1 -> x2 = y2 -> x3 = y3 -> f x1 x2 x3 = f y1 y2 y3.
+
+.. _datatypes:
+
+Datatypes
+~~~~~~~~~
+
+.. index::
+   single: Datatypes
+
+In the basic library, we find in ``Datatypes.v`` the definition
+of the basic data-types of programming,
+defined as inductive constructions over the sort ``Set``. Some of
+them come with a special syntax shown below (this syntax table is common with
+the next section :ref:`specification`):
+
+.. productionlist::
+   specif :  `specif` * `specif`                           (prod)
+          : | `specif` + `specif`                          (sum)
+          : | `specif` + { `specif` }                      (sumor)
+          : | { `specif` } + { `specif` }                  (sumbool)
+          : | { `ident` : `specif` | `form` }              (sig)
+          : | { `ident` : `specif` | `form` & `form` }       (sig2)
+          : | { `ident` : `specif` & `specif` }             (sigT)
+          : | { `ident` : `specif` & `specif` & `specif` }    (sigT2)
+  term : (`term`, `term`)                               (pair)
```

Comment:
Doesn't look good to me either:
![screenshot-2018-4-3 the coq library coq 8 8 beta1 documentation](https://user-images.githubusercontent.com/1108325/38251595-31e7ef1a-3752-11e8-9a4e-3867554177ff.png)